### DISCUSSION TITLE
Render content item base on contenttype definition ?

### DISCUSSION THREAD

**Discussion by orchardc** [OrchardCMS/OrchardCore#13308] [Q&A]
Create a simple ContentType call "TEST", with parts "Title" and "HtmlBody" included.

![image](https://user-images.githubusercontent.com/16193516/221408119-f6cc20ee-7505-4e30-a2b3-ad69671c4193.png)

How to render this ContentItem in MVC/Liquid? 

The goal is to allow Internet users to create content items based on the "TEST" definition in a webpage e..g  ../TEST/create.

It would be good to have an example.

Many thanks!

**Comment by hishamco** [comment]
There's a lot of valuable resources on our [docs](http://docs.orchardcore.net/)

**Reply by orchardc** [reply]
I know the doc but could not find a specific example for it...  I must had missed something .

**Reply by hishamco** [reply]
```
@{
    var articles = await Orchard.GetRecentContentItemsByContentTypeAsync("Articles");
}
```

Then you can loop through your content items. FYI there're a lot of helper method to get content item(s) by id, alias .. etc

**Reply by orchardc** [reply]
Thanks Hisham,

The idea is every time the contenttype is updated on the backend(through the Admin site), for example , adding a new Date part call Birthday. There is no need to update the front part UI , so the new field should be rendered automatically when users creating a new content item based on that contenttype.

I hope it make sense?

So the front-end page (MVC or liquid) needs to be a general page to render whatever defined for that content type.

**Reply by hishamco** [reply]
OC by default generate a view shows all the fields for a particular content type. Try to view a particular content item from the admin

**Reply by orchardc** [reply]
Yes but that's the Admin view and I try to avoid using it,  instead I try to create a liquid/MVC view that has more control on UI.

Is there no way to list all fields of a ContendType on a liquid/MVC view?

**Reply by hishamco** [reply]
You can override the default view if you do so with the help of placement and shapes

**Reply by orchardc** [reply]
Do you mean to overwrite this page?
![image](https://user-images.githubusercontent.com/16193516/221419976-8c4f61f8-4e14-4830-ac89-cbc997363a3c.png)

the URL is ../Admin/Contents/ContentTypes/TEST/Create,
I'm thinking there must an API/method on that Create page, which can do:
1. List all content fields/part of this ContentType
2. When click 'Publish', then create a new content item.

There must be an easy way to do above #1 and #2...

**Reply by hishamco** [reply]
@orchardc could you please write down the steps to reproduce the issue, seems I got lost :)

**Reply by orchardc** [reply]
Sorry Hisham, I must confuse you on this.

All I'm saying is, I'd expect a simple method, which can list all content fields/parts of a given ContentType (e.g. TEST in my example) on a MVC/Liquid page, so an Internet user will be able to create/update his content item without using (overwriting) the Admin page.

**Reply by hishamco** [reply]
If you need to do the same experience on the front-end I might suggest to look into the content type admin views
